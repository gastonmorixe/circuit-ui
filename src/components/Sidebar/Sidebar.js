import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';

import Header from './components/Header';
import NavList from './components/NavList';
import NavItem from './components/NavItem';
import Backdrop from './components/Backdrop';
import CloseButton from './components/CloseButton';

const SIDEBAR_WIDTH = 256;

const baseStyles = ({ theme, open }) => css`
  label: sidebar;
  display: flex;
  flex-direction: column;
  height: 100%;
  width: ${SIDEBAR_WIDTH}px;
  background-color: ${theme.colors.n900};
  transition: transform 150ms ease-in-out;
  position: absolute;
  transform: translateX(${open ? 0 : `-${SIDEBAR_WIDTH}px`});
  z-index: 1;
  ${theme.mq.mega`
    transform: translateX(0);
    position: relative;
  `};
`;

const Drawer = styled('div')(baseStyles);

const Sidebar = ({ children, open, onClose }) => (
  <Fragment>
    <Drawer open={open}>{children}</Drawer>
    <Backdrop visible={open} />
    <CloseButton visible={open} onClick={onClose} />
  </Fragment>
);

Sidebar.propTypes = {
  /**
   * The children component passed to the Sidebar
   */
  children: PropTypes.node,
  /**
   * Tells if the Sidebar is open
   */
  open: PropTypes.bool,
  /**
   * A function to handle the sidebar close
   */
  onClose: PropTypes.func
};

Sidebar.Header = Header;
Sidebar.NavList = NavList;
Sidebar.NavItem = NavItem;

export default Sidebar;
