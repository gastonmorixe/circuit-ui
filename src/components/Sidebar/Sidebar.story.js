import React, { Component } from 'react';
import { range } from 'lodash/fp';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import { GROUPS } from '../../../.storybook/hierarchySeparators';
import withTests from '../../util/withTests';
import Sidebar from '.';

class SidebarContainer extends Component {
  state = { selectedItem: 0 };

  handleNavItemClick = index => {
    this.setState({ selectedItem: index });
  };

  render() {
    return (
      <div style={{ height: '600px' }}>
        <Sidebar>
          <Sidebar.Header>Header</Sidebar.Header>
          <Sidebar.NavList>
            {range(1, 5).map((value, i) => (
              <Sidebar.NavItem
                key={i}
                selected={this.state.selectedItem === i}
                onClick={() => this.handleNavItemClick(i)}
              >
                Item #{i}
              </Sidebar.NavItem>
            ))}
          </Sidebar.NavList>
        </Sidebar>
      </div>
    );
  }
}

storiesOf(`${GROUPS.COMPONENTS}|Sidebar`, module)
  .addDecorator(withTests('Sidebar'))
  .add('Sidebar', withInfo()(() => <SidebarContainer />));
