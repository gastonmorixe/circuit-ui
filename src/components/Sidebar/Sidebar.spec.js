import React from 'react';
import Sidebar from './Sidebar';

describe('<Sidebar />', () => {
  it('should render and match the snapshot', () => {
    const actual = mount(<Sidebar />);
    expect(actual).toMatchSnapshot();
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(<Sidebar />);
      const actual = await axe(wrapper);
      expect(actual).toHaveNoViolations();
    });
  });
});
