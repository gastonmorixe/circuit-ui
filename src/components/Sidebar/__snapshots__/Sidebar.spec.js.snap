// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should render and match snapshot when open 1`] = `
Array [
  .circuit-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 256px;
  background-color: #212933;
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
  position: absolute;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  z-index: 1;
}

@media (min-width:768px) {
  .circuit-0 {
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    position: relative;
  }
}

<div
    className="circuit-0 circuit-1"
    open={true}
  />,
  .circuit-0 {
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: black;
  -webkit-transition: opacity 200ms ease-in-out,visibility 200ms ease-in-out;
  transition: opacity 200ms ease-in-out,visibility 200ms ease-in-out;
  visibility: visible;
  opacity: 0.56;
  z-index: 0;
}

@media (min-width:768px) {
  .circuit-0 {
    visibility: hidden;
  }
}

<div
    className="circuit-0 circuit-1"
  />,
  .circuit-0 {
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  background-color: #FAFBFC;
  position: absolute;
  bottom: 16px;
  right: 16px;
  -webkit-transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  visibility: visible;
  opacity: 1;
  z-index: 1;
}

@media (min-width:768px) {
  .circuit-0 {
    visibility: hidden;
  }
}

<button
    className="circuit-0 circuit-1"
    onClick={[MockFunction]}
  >
    <div>
      closeIcon.svg
    </div>
  </button>,
]
`;

exports[`<Sidebar /> should render and match the snapshot when closed 1`] = `
Array [
  .circuit-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 256px;
  background-color: #212933;
  -webkit-transition: -webkit-transform 200ms ease-in-out;
  -webkit-transition: transform 200ms ease-in-out;
  transition: transform 200ms ease-in-out;
  position: absolute;
  -webkit-transform: translateX(-256px);
  -ms-transform: translateX(-256px);
  transform: translateX(-256px);
  z-index: 1;
}

@media (min-width:768px) {
  .circuit-0 {
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    position: relative;
  }
}

<div
    className="circuit-0 circuit-1"
    open={false}
  />,
  .circuit-0 {
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: black;
  -webkit-transition: opacity 200ms ease-in-out,visibility 200ms ease-in-out;
  transition: opacity 200ms ease-in-out,visibility 200ms ease-in-out;
  visibility: hidden;
  opacity: 0;
  z-index: 0;
}

@media (min-width:768px) {
  .circuit-0 {
    visibility: hidden;
  }
}

<div
    className="circuit-0 circuit-1"
  />,
  .circuit-0 {
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  background-color: #FAFBFC;
  position: absolute;
  bottom: 16px;
  right: 16px;
  -webkit-transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  transition: opacity 150ms ease-in-out,visibility 150ms ease-in-out;
  visibility: hidden;
  opacity: 0;
  z-index: 1;
}

@media (min-width:768px) {
  .circuit-0 {
    visibility: hidden;
  }
}

<button
    className="circuit-0 circuit-1"
    onClick={[MockFunction]}
  >
    <div>
      closeIcon.svg
    </div>
  </button>,
]
`;
